src/content/sitemap.rs

```diff
@@ -23,40 +23,38 @@ use super::{get_content, mime_type::Mime, Content};
 pub struct ContentTree
 {
     uri_stem: String,
-    contents: Vec<(Content, Arc<Mutex<bool>>)>,
-    children: BTreeMap<String, ContentTree>
+    contents: BTreeMap<String, Arc<Mutex<Content>>>,
+    content_types: BTreeMap<String, MIME>,
+    children: BTreeMap<String, ContentTree>,
+    sitmap_content: bool
 }
 
 impl ContentTree
 {
     pub fn new(uri_stem: &str) -> ContentTree
     {
-        ContentTree { uri_stem: uri_stem.to_string(), contents: vec![], children: BTreeMap::new() }
+        ContentTree { uri_stem: uri_stem.to_string(), contents: BTreeMap::new(), children: BTreeMap::new(), sitmap_content: false, content_types: BTreeMap::new() }
     }
 
+    /// Build a [Router] to serve the content. If static_router then
+    ///   content is never refreshed within the router
     fn route(&self, static_router: bool) -> Router
     {
         let mut router: Router<(), axum::body::Body> = Router::new();
-        for (mut content, mutex) in self.contents.clone()
+        for (uri, content) in self.contents.clone()
         {
-            content.refresh();
             router = router.route
             (
-                &content.get_uri(), 
+                &uri, 
                 get(move || async move 
                     {
-                        // check if we should attempt a lock
+                        let mut content = content.lock().await;
                         if !static_router && content.server_cache_expired() && content.is_stale()
                         {
-                            let _ = mutex.lock().await;
-                            if content.server_cache_expired() && content.is_stale()
-                            {
-                                // got the lock, and still stale
-                                content.refresh();
-                                crate::debug(format!("Refresh called on Content {}", content.get_uri()), None);
-                            }
+                            content.refresh();
+                            crate::debug(format!("Refresh called on Content {}", content.get_uri()), None);
                         }
-                        content.into_response()
+                        content.clone().into_response()
                     })
             );
         }
```

Comment:
Benchmark with heaptrack compared to main

Reply:
| main  | pr |
| ------------- | ------------- |
| ![Screenshot from 2024-06-04 21-29-36](https://github.com/JerboaBurrow/Busser/assets/84378622/3714875a-7040-4fd9-b692-5ac39831c94a)  | ![Screenshot from 2024-06-04 21-30-18](https://github.com/JerboaBurrow/Busser/assets/84378622/6e1ee40c-5df7-46c9-b97d-081be46b27c4) |
| ![Screenshot from 2024-06-04 21-29-58](https://github.com/JerboaBurrow/Busser/assets/84378622/0aada3f7-4f39-413d-a8a4-f1f52ed311af)  | ![Screenshot from 2024-06-04 21-30-29](https://github.com/JerboaBurrow/Busser/assets/84378622/8d2ec0eb-cf99-4800-beba-a886c4d5f415)  |